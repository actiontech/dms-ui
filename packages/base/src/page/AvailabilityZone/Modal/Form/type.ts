import { FormInstance } from 'antd';

export type AvailabilityZoneFormType = {
  name: string;
  address: string;
};

export type AvailabilityZoneFormProps = {
  form: FormInstance<AvailabilityZoneFormType>;
  isUpdate?: boolean;
};
