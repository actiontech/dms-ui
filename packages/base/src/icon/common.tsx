export const IconCommonPassword: React.FC = () => {
  return (
    <svg
      width="18"
      height="19"
      viewBox="0 0 18 19"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g id="lock-password-fill">
        <path
          id="Vector"
          d="M13.5 6.5H15C15.1989 6.5 15.3897 6.57902 15.5303 6.71967C15.671 6.86032 15.75 7.05109 15.75 7.25V16.25C15.75 16.4489 15.671 16.6397 15.5303 16.7803C15.3897 16.921 15.1989 17 15 17H3C2.80109 17 2.61032 16.921 2.46967 16.7803C2.32902 16.6397 2.25 16.4489 2.25 16.25V7.25C2.25 7.05109 2.32902 6.86032 2.46967 6.71967C2.61032 6.57902 2.80109 6.5 3 6.5H4.5V5.75C4.5 4.55653 4.97411 3.41193 5.81802 2.56802C6.66193 1.72411 7.80653 1.25 9 1.25C10.1935 1.25 11.3381 1.72411 12.182 2.56802C13.0259 3.41193 13.5 4.55653 13.5 5.75V6.5ZM12 6.5V5.75C12 4.95435 11.6839 4.19129 11.1213 3.62868C10.5587 3.06607 9.79565 2.75 9 2.75C8.20435 2.75 7.44129 3.06607 6.87868 3.62868C6.31607 4.19129 6 4.95435 6 5.75V6.5H12ZM8.25 11V12.5H9.75V11H8.25ZM5.25 11V12.5H6.75V11H5.25ZM11.25 11V12.5H12.75V11H11.25Z"
          fill="#C3C6CD"
        />
      </g>
    </svg>
  );
};

export const IconCommonUser: React.FC = () => {
  return (
    <svg
      width="18"
      height="19"
      viewBox="0 0 18 19"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g id="user-4-fill">
        <g id="Vector">
          <path
            d="M6.47462 7.34681C5.80485 6.67704 5.42857 5.76863 5.42857 4.82143C5.42857 3.87423 5.80485 2.96582 6.47462 2.29605C7.14439 1.62627 8.0528 1.25 9 1.25C9.9472 1.25 10.8556 1.62627 11.5254 2.29605C12.1952 2.96582 12.5714 3.87423 12.5714 4.82143C12.5714 5.76863 12.1952 6.67704 11.5254 7.34681C10.8556 8.01658 9.9472 8.39286 9 8.39286C8.0528 8.39286 7.14439 8.01658 6.47462 7.34681Z"
            fill="#C3C6CD"
          />
          <path
            d="M16.5 15.1785C16.5 12.0225 13.9418 9.46426 10.7857 9.46426H7.21429C4.05821 9.46426 1.5 12.0225 1.5 15.1785V15.5357C1.5 15.9146 1.65051 16.2779 1.91842 16.5458C2.18633 16.8137 2.54969 16.9643 2.92857 16.9643H15.0714C15.4503 16.9643 15.8137 16.8137 16.0816 16.5458C16.3495 16.2779 16.5 15.9146 16.5 15.5357V15.1785Z"
            fill="#C3C6CD"
          />
        </g>
      </g>
    </svg>
  );
};
