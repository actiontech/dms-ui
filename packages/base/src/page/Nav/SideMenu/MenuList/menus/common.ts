import { UserRolesType } from '@actiontech/shared/lib/enum';

import {
  CustomMenuItemType,
  GenerateMenuItemType,
  MenuStructTreeKey,
  MenuStructTreeType
} from './index.type';

export const SIDE_MENU_DATA_PLACEHOLDER_KEY = 'projectID';

export const genMenuItemsWithMenuStructTree = (
  projectID: string,
  allMenuItems: GenerateMenuItemType[],
  menuStructTree: MenuStructTreeType,
  userRoles: UserRolesType
): CustomMenuItemType[] => {
  const getMenuItemWithKey = (key: MenuStructTreeKey): CustomMenuItemType => {
    return (
      allMenuItems.find((v) => {
        const menu = v(projectID);
        if (menu?.role) {
          return (
            menu.role.some((role) => userRoles[role]) && menu.structKey === key
          );
        }

        return menu?.structKey === key;
      })?.(projectID) ?? null
    );
  };

  return menuStructTree.map<CustomMenuItemType>((item) => {
    if (typeof item === 'string') {
      return getMenuItemWithKey(item);
    }

    if (item.type === 'group') {
      const children = item.group.map(getMenuItemWithKey);
      if (children.every((v) => v === null)) {
        return null;
      }
      return {
        type: 'group',
        label: item.label,
        children
      } as CustomMenuItemType;
    }

    return item as CustomMenuItemType;
  });
};
