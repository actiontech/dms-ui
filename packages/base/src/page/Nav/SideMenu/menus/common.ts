import { MenuProps } from 'antd';
import { ItemType, SubMenuType } from 'antd/es/menu/hooks/useItems';
import { NavigateFunction } from 'react-router-dom';

export type CustomMenuItemType =
  | (ItemType & {
      order?: number;
    })
  | null;

export type GenerateMenuItemsType = (arg: {
  navigate: NavigateFunction;
  projectID?: string;
}) => CustomMenuItemType[];

export const SIDE_MENU_DATA_PLACEHOLDER_KEY = 'projectID';

export const isAdminKeys = {
  operate: `sqle/project/${SIDE_MENU_DATA_PLACEHOLDER_KEY}/operationRecord`
} as const;

export const filterAdminMenusWithKey = (
  menus: MenuProps['items']
): CustomMenuItemType[] => {
  return (
    menus?.filter((v) => {
      const menu = v as SubMenuType;
      if (menu.children) {
        menu.children = filterAdminMenusWithKey(menu.children) ?? [];
      }

      return Object.keys(isAdminKeys).every((e) => {
        const isAdminKey = e as keyof typeof isAdminKeys;
        return isAdminKeys[isAdminKey] !== menu?.key;
      });
    }) ?? []
  );
};
