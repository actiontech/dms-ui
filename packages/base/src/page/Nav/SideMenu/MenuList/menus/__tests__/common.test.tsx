import { PERMISSIONS } from '@actiontech/shared/lib/global';
import { t } from '../../../../../../locale';
import baseMenusCollection from '../base';
import { genMenuItemsWithMenuStructTree } from '../common';
import { MenuStructTreeType } from '../index.type';
import sqleMenusCollection from '../sqle';

const menuStruct: MenuStructTreeType = [
  'project-overview',
  { type: 'divider' },
  {
    type: 'group',
    label: t('dmsMenu.groupLabel.SQLDev'),
    group: [
      'cloud-beaver',
      'data-export',
      'sql-audit',
      'plugin-audit',
      'sql-optimization',
      'data-source-comparison'
    ]
  },
  {
    type: 'group',
    label: t('dmsMenu.groupLabel.SQLExecute'),
    group: ['exec-workflow', 'version-management']
  },
  {
    type: 'group',
    label: t('dmsMenu.groupLabel.CICDIntegration'),
    group: ['pipeline-configuration']
  },
  {
    type: 'group',
    label: t('dmsMenu.groupLabel.SQLManagement'),
    group: ['sql-management', 'sql-management-conf']
  },
  { type: 'divider' },
  {
    type: 'group',
    label: t('dmsMenu.groupLabel.projectConfigure'),
    group: [
      'instance',
      'rule-template',
      'workflow-template',
      'member',
      'push-rule-configuration',
      'whitelist',
      'sql-management-exception'
    ]
  },
  { type: 'divider' },
  {
    type: 'group',
    permission: PERMISSIONS.PAGES.SQLE.OPERATION_RECORD,
    label: t('dmsMenu.groupLabel.operateAndAudit'),
    group: ['sqle-log']
  }
];

describe('test genMenuItemsWithMenuStructTree', () => {
  it('should match snapshot', () => {
    expect(genMenuItemsWithMenuStructTree('600300', [], [])).toMatchSnapshot();
    expect(
      genMenuItemsWithMenuStructTree('600300', baseMenusCollection, [])
    ).toMatchSnapshot();

    expect(
      genMenuItemsWithMenuStructTree('600300', baseMenusCollection, [])
    ).toMatchSnapshot();

    expect(
      genMenuItemsWithMenuStructTree('600300', baseMenusCollection, menuStruct)
    ).toMatchSnapshot();

    expect(
      genMenuItemsWithMenuStructTree('600300', baseMenusCollection, menuStruct)
    ).toMatchSnapshot();

    expect(
      genMenuItemsWithMenuStructTree(
        '600300',
        [...baseMenusCollection, ...sqleMenusCollection],
        menuStruct
      )
    ).toMatchSnapshot();

    expect(
      genMenuItemsWithMenuStructTree(
        '600300',
        [...baseMenusCollection, ...sqleMenusCollection],
        menuStruct
      )
    ).toMatchSnapshot();
  });
});
