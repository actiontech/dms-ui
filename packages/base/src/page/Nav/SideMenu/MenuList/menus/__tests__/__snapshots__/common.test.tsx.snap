// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test genMenuItemsWithMenuStructTree should match snapshot 1`] = `[]`;

exports[`test genMenuItemsWithMenuStructTree should match snapshot 2`] = `[]`;

exports[`test genMenuItemsWithMenuStructTree should match snapshot 3`] = `[]`;

exports[`test genMenuItemsWithMenuStructTree should match snapshot 4`] = `
[
  null,
  {
    "type": "divider",
  },
  {
    "children": [
      {
        "icon": <ComputerFilled
          height={18}
          width={18}
        />,
        "key": "project/projectID/cloud-beaver",
        "label": <ForwardRef(TypedLink)
          params={
            {
              "projectID": "600300",
            }
          }
          to={
            {
              "path": ":projectID/cloud-beaver",
              "prefix": "/project",
              "query": "open_cloud_beaver",
            }
          }
        >
          SQL工作台
        </ForwardRef(TypedLink)>,
        "structKey": "cloud-beaver",
      },
      {
        "icon": <ExportFilled
          height={18}
          width={18}
        />,
        "key": "project/projectID/data/export",
        "label": <ForwardRef(TypedLink)
          params={
            {
              "projectID": "600300",
            }
          }
          to={
            {
              "path": ":projectID/data/export",
              "prefix": "/project",
            }
          }
        >
          数据导出
        </ForwardRef(TypedLink)>,
        "structKey": "data-export",
      },
      null,
      null,
      null,
      null,
    ],
    "label": "SQL开发",
    "permission": undefined,
    "type": "group",
  },
  null,
  null,
  null,
  {
    "type": "divider",
  },
  {
    "children": [
      {
        "icon": <DatabaseFilled
          height={18}
          width={18}
        />,
        "key": "project/projectID/db-services",
        "label": <ForwardRef(TypedLink)
          params={
            {
              "projectID": "600300",
            }
          }
          to={
            {
              "path": ":projectID/db-services",
              "prefix": "/project",
              "query": "address",
            }
          }
        >
          数据源
        </ForwardRef(TypedLink)>,
        "structKey": "instance",
      },
      null,
      null,
      {
        "icon": <MemberFilled
          height={18}
          width={18}
        />,
        "key": "project/projectID/member",
        "label": <ForwardRef(TypedLink)
          params={
            {
              "projectID": "600300",
            }
          }
          to={
            {
              "path": ":projectID/member",
              "prefix": "/project",
            }
          }
        >
          成员与权限
        </ForwardRef(TypedLink)>,
        "structKey": "member",
      },
      null,
      null,
      null,
    ],
    "label": "项目配置",
    "permission": undefined,
    "type": "group",
  },
  {
    "type": "divider",
  },
  null,
]
`;

exports[`test genMenuItemsWithMenuStructTree should match snapshot 5`] = `
[
  null,
  {
    "type": "divider",
  },
  {
    "children": [
      {
        "icon": <ComputerFilled
          height={18}
          width={18}
        />,
        "key": "project/projectID/cloud-beaver",
        "label": <ForwardRef(TypedLink)
          params={
            {
              "projectID": "600300",
            }
          }
          to={
            {
              "path": ":projectID/cloud-beaver",
              "prefix": "/project",
              "query": "open_cloud_beaver",
            }
          }
        >
          SQL工作台
        </ForwardRef(TypedLink)>,
        "structKey": "cloud-beaver",
      },
      {
        "icon": <ExportFilled
          height={18}
          width={18}
        />,
        "key": "project/projectID/data/export",
        "label": <ForwardRef(TypedLink)
          params={
            {
              "projectID": "600300",
            }
          }
          to={
            {
              "path": ":projectID/data/export",
              "prefix": "/project",
            }
          }
        >
          数据导出
        </ForwardRef(TypedLink)>,
        "structKey": "data-export",
      },
      null,
      null,
      null,
      null,
    ],
    "label": "SQL开发",
    "permission": undefined,
    "type": "group",
  },
  null,
  null,
  null,
  {
    "type": "divider",
  },
  {
    "children": [
      {
        "icon": <DatabaseFilled
          height={18}
          width={18}
        />,
        "key": "project/projectID/db-services",
        "label": <ForwardRef(TypedLink)
          params={
            {
              "projectID": "600300",
            }
          }
          to={
            {
              "path": ":projectID/db-services",
              "prefix": "/project",
              "query": "address",
            }
          }
        >
          数据源
        </ForwardRef(TypedLink)>,
        "structKey": "instance",
      },
      null,
      null,
      {
        "icon": <MemberFilled
          height={18}
          width={18}
        />,
        "key": "project/projectID/member",
        "label": <ForwardRef(TypedLink)
          params={
            {
              "projectID": "600300",
            }
          }
          to={
            {
              "path": ":projectID/member",
              "prefix": "/project",
            }
          }
        >
          成员与权限
        </ForwardRef(TypedLink)>,
        "structKey": "member",
      },
      null,
      null,
      null,
    ],
    "label": "项目配置",
    "permission": undefined,
    "type": "group",
  },
  {
    "type": "divider",
  },
  null,
]
`;

exports[`test genMenuItemsWithMenuStructTree should match snapshot 6`] = `
[
  {
    "icon": <OverviewOutlined
      height={18}
      width={18}
    />,
    "key": "sqle/project/projectID/overview",
    "label": <ForwardRef(TypedLink)
      params={
        {
          "projectID": "600300",
        }
      }
      to={
        {
          "path": ":projectID/overview",
          "prefix": "/sqle/project",
        }
      }
    >
      项目概览
    </ForwardRef(TypedLink)>,
    "structKey": "project-overview",
  },
  {
    "type": "divider",
  },
  {
    "children": [
      {
        "icon": <ComputerFilled
          height={18}
          width={18}
        />,
        "key": "project/projectID/cloud-beaver",
        "label": <ForwardRef(TypedLink)
          params={
            {
              "projectID": "600300",
            }
          }
          to={
            {
              "path": ":projectID/cloud-beaver",
              "prefix": "/project",
              "query": "open_cloud_beaver",
            }
          }
        >
          SQL工作台
        </ForwardRef(TypedLink)>,
        "structKey": "cloud-beaver",
      },
      {
        "icon": <ExportFilled
          height={18}
          width={18}
        />,
        "key": "project/projectID/data/export",
        "label": <ForwardRef(TypedLink)
          params={
            {
              "projectID": "600300",
            }
          }
          to={
            {
              "path": ":projectID/data/export",
              "prefix": "/project",
            }
          }
        >
          数据导出
        </ForwardRef(TypedLink)>,
        "structKey": "data-export",
      },
      {
        "icon": <MagnifierFilled
          height={18}
          width={18}
        />,
        "key": "sqle/project/projectID/sql-audit",
        "label": <ForwardRef(TypedLink)
          params={
            {
              "projectID": "600300",
            }
          }
          to={
            {
              "path": ":projectID/sql-audit",
              "prefix": "/sqle/project",
              "query": "SQLAuditRecordID",
            }
          }
        >
          快捷审核
        </ForwardRef(TypedLink)>,
        "structKey": "sql-audit",
      },
      {
        "icon": <CodeOutlined
          height={18}
          width={18}
        />,
        "key": "sqle/project/projectID/plugin-audit",
        "label": <ForwardRef(TypedLink)
          params={
            {
              "projectID": "600300",
            }
          }
          to={
            {
              "path": ":projectID/plugin-audit",
              "prefix": "/sqle/project",
            }
          }
        >
          IDE审核
        </ForwardRef(TypedLink)>,
        "structKey": "plugin-audit",
      },
      {
        "icon": <RiseSquareOutlined
          height={18}
          width={18}
        />,
        "key": "sqle/project/projectID/sql-optimization",
        "label": <ForwardRef(TypedLink)
          params={
            {
              "projectID": "600300",
            }
          }
          to={
            {
              "path": ":projectID/sql-optimization",
              "prefix": "/sqle/project",
            }
          }
        >
          智能调优
        </ForwardRef(TypedLink)>,
        "permission": "page:sql_optimization",
        "structKey": "sql-optimization",
      },
      {
        "icon": <ComparisonOutlined
          height={18}
          width={18}
        />,
        "key": "sqle/project/projectID/data-source-comparison",
        "label": <ForwardRef(TypedLink)
          params={
            {
              "projectID": "600300",
            }
          }
          to={
            {
              "path": ":projectID/data-source-comparison",
              "prefix": "/sqle/project",
            }
          }
        >
          结构对比
        </ForwardRef(TypedLink)>,
        "structKey": "data-source-comparison",
      },
    ],
    "label": "SQL开发",
    "permission": undefined,
    "type": "group",
  },
  {
    "children": [
      {
        "icon": <BriefcaseFilled
          height={18}
          width={18}
        />,
        "key": "sqle/project/projectID/exec-workflow",
        "label": <ForwardRef(TypedLink)
          params={
            {
              "projectID": "600300",
            }
          }
          to={
            {
              "path": ":projectID/exec-workflow",
              "prefix": "/sqle/project",
            }
          }
        >
          SQL工单
        </ForwardRef(TypedLink)>,
        "structKey": "exec-workflow",
      },
      {
        "icon": <FileVersionOutlined
          height={18}
          width={18}
        />,
        "key": "sqle/project/projectID/version-management",
        "label": <ForwardRef(TypedLink)
          params={
            {
              "projectID": "600300",
            }
          }
          to={
            {
              "path": ":projectID/version-management",
              "prefix": "/sqle/project",
            }
          }
        >
          版本管理
        </ForwardRef(TypedLink)>,
        "structKey": "version-management",
      },
    ],
    "label": "SQL上线",
    "permission": undefined,
    "type": "group",
  },
  {
    "children": [
      {
        "icon": <PipelineOutlined
          height={18}
          width={18}
        />,
        "key": "sqle/project/projectID/pipeline-configuration",
        "label": <ForwardRef(TypedLink)
          params={
            {
              "projectID": "600300",
            }
          }
          to={
            {
              "path": ":projectID/pipeline-configuration",
              "prefix": "/sqle/project",
            }
          }
        >
          流水线配置
        </ForwardRef(TypedLink)>,
        "structKey": "pipeline-configuration",
      },
    ],
    "label": "CI/CD集成",
    "permission": undefined,
    "type": "group",
  },
  {
    "children": [
      {
        "icon": <ManagementFilled
          height={18}
          width={18}
        />,
        "key": "sqle/project/projectID/sql-management",
        "label": <ForwardRef(TypedLink)
          params={
            {
              "projectID": "600300",
            }
          }
          to={
            {
              "path": ":projectID/sql-management",
              "prefix": "/sqle/project",
              "query": "instance_id&source",
            }
          }
        >
          SQL管控
        </ForwardRef(TypedLink)>,
        "structKey": "sql-management",
      },
      {
        "icon": <PlanFilled
          height={18}
          width={18}
        />,
        "key": "sqle/project/projectID/sql-management-conf",
        "label": <ForwardRef(TypedLink)
          params={
            {
              "projectID": "600300",
            }
          }
          to={
            {
              "path": ":projectID/sql-management-conf",
              "prefix": "/sqle/project",
            }
          }
        >
          SQL管控配置
        </ForwardRef(TypedLink)>,
        "structKey": "sql-management-conf",
      },
    ],
    "label": "SQL管控",
    "permission": undefined,
    "type": "group",
  },
  {
    "type": "divider",
  },
  {
    "children": [
      {
        "icon": <DatabaseFilled
          height={18}
          width={18}
        />,
        "key": "project/projectID/db-services",
        "label": <ForwardRef(TypedLink)
          params={
            {
              "projectID": "600300",
            }
          }
          to={
            {
              "path": ":projectID/db-services",
              "prefix": "/project",
              "query": "address",
            }
          }
        >
          数据源
        </ForwardRef(TypedLink)>,
        "structKey": "instance",
      },
      {
        "icon": <ProfileSquareFilled
          fill="none"
          height={18}
          width={18}
        />,
        "key": "sqle/project/projectID/rule/template",
        "label": <ForwardRef(TypedLink)
          params={
            {
              "projectID": "600300",
            }
          }
          to={
            {
              "path": ":projectID/rule/template",
              "prefix": "/sqle/project",
            }
          }
        >
          审核规则模板
        </ForwardRef(TypedLink)>,
        "structKey": "rule-template",
      },
      {
        "icon": <WorkflowFilled
          height={18}
          width={18}
        />,
        "key": "sqle/project/projectID/progress",
        "label": <ForwardRef(TypedLink)
          params={
            {
              "projectID": "600300",
            }
          }
          to={
            {
              "path": ":projectID/progress",
              "prefix": "/sqle/project",
            }
          }
        >
          审批流程模板
        </ForwardRef(TypedLink)>,
        "structKey": "workflow-template",
      },
      {
        "icon": <MemberFilled
          height={18}
          width={18}
        />,
        "key": "project/projectID/member",
        "label": <ForwardRef(TypedLink)
          params={
            {
              "projectID": "600300",
            }
          }
          to={
            {
              "path": ":projectID/member",
              "prefix": "/project",
            }
          }
        >
          成员与权限
        </ForwardRef(TypedLink)>,
        "structKey": "member",
      },
      {
        "icon": <GearFilled
          height={18}
          width={18}
        />,
        "key": "sqle/project/projectID/push-rule",
        "label": <ForwardRef(TypedLink)
          params={
            {
              "projectID": "600300",
            }
          }
          to={
            {
              "path": ":projectID/push-rule",
              "prefix": "/sqle/project",
            }
          }
        >
          推送规则
        </ForwardRef(TypedLink)>,
        "structKey": "push-rule-configuration",
      },
      {
        "icon": <ResolveFileFilled
          height={18}
          width={18}
        />,
        "key": "sqle/project/projectID/whitelist",
        "label": <ForwardRef(TypedLink)
          params={
            {
              "projectID": "600300",
            }
          }
          to={
            {
              "path": ":projectID/whitelist",
              "prefix": "/sqle/project",
            }
          }
        >
          审核SQL例外
        </ForwardRef(TypedLink)>,
        "structKey": "whitelist",
      },
      {
        "icon": <ExceptionFileOutlined
          height={18}
          width={18}
        />,
        "key": "sqle/project/projectID/sql-management-exception",
        "label": <ForwardRef(TypedLink)
          params={
            {
              "projectID": "600300",
            }
          }
          to={
            {
              "path": ":projectID/sql-management-exception",
              "prefix": "/sqle/project",
            }
          }
        >
          管控SQL例外
        </ForwardRef(TypedLink)>,
        "structKey": "sql-management-exception",
      },
    ],
    "label": "项目配置",
    "permission": undefined,
    "type": "group",
  },
  {
    "type": "divider",
  },
  {
    "children": [
      {
        "icon": <OperateAuditFilled
          height={18}
          width={18}
        />,
        "key": "sqle/project/projectID/operation-record",
        "label": <ForwardRef(TypedLink)
          params={
            {
              "projectID": "600300",
            }
          }
          to={
            {
              "path": ":projectID/operation-record",
              "prefix": "/sqle/project",
            }
          }
        >
          SQLE操作记录
        </ForwardRef(TypedLink)>,
        "permission": "page:operation_record",
        "structKey": "sqle-log",
      },
    ],
    "label": "操作与审计",
    "permission": "page:operation_record",
    "type": "group",
  },
]
`;

exports[`test genMenuItemsWithMenuStructTree should match snapshot 7`] = `
[
  {
    "icon": <OverviewOutlined
      height={18}
      width={18}
    />,
    "key": "sqle/project/projectID/overview",
    "label": <ForwardRef(TypedLink)
      params={
        {
          "projectID": "600300",
        }
      }
      to={
        {
          "path": ":projectID/overview",
          "prefix": "/sqle/project",
        }
      }
    >
      项目概览
    </ForwardRef(TypedLink)>,
    "structKey": "project-overview",
  },
  {
    "type": "divider",
  },
  {
    "children": [
      {
        "icon": <ComputerFilled
          height={18}
          width={18}
        />,
        "key": "project/projectID/cloud-beaver",
        "label": <ForwardRef(TypedLink)
          params={
            {
              "projectID": "600300",
            }
          }
          to={
            {
              "path": ":projectID/cloud-beaver",
              "prefix": "/project",
              "query": "open_cloud_beaver",
            }
          }
        >
          SQL工作台
        </ForwardRef(TypedLink)>,
        "structKey": "cloud-beaver",
      },
      {
        "icon": <ExportFilled
          height={18}
          width={18}
        />,
        "key": "project/projectID/data/export",
        "label": <ForwardRef(TypedLink)
          params={
            {
              "projectID": "600300",
            }
          }
          to={
            {
              "path": ":projectID/data/export",
              "prefix": "/project",
            }
          }
        >
          数据导出
        </ForwardRef(TypedLink)>,
        "structKey": "data-export",
      },
      {
        "icon": <MagnifierFilled
          height={18}
          width={18}
        />,
        "key": "sqle/project/projectID/sql-audit",
        "label": <ForwardRef(TypedLink)
          params={
            {
              "projectID": "600300",
            }
          }
          to={
            {
              "path": ":projectID/sql-audit",
              "prefix": "/sqle/project",
              "query": "SQLAuditRecordID",
            }
          }
        >
          快捷审核
        </ForwardRef(TypedLink)>,
        "structKey": "sql-audit",
      },
      {
        "icon": <CodeOutlined
          height={18}
          width={18}
        />,
        "key": "sqle/project/projectID/plugin-audit",
        "label": <ForwardRef(TypedLink)
          params={
            {
              "projectID": "600300",
            }
          }
          to={
            {
              "path": ":projectID/plugin-audit",
              "prefix": "/sqle/project",
            }
          }
        >
          IDE审核
        </ForwardRef(TypedLink)>,
        "structKey": "plugin-audit",
      },
      {
        "icon": <RiseSquareOutlined
          height={18}
          width={18}
        />,
        "key": "sqle/project/projectID/sql-optimization",
        "label": <ForwardRef(TypedLink)
          params={
            {
              "projectID": "600300",
            }
          }
          to={
            {
              "path": ":projectID/sql-optimization",
              "prefix": "/sqle/project",
            }
          }
        >
          智能调优
        </ForwardRef(TypedLink)>,
        "permission": "page:sql_optimization",
        "structKey": "sql-optimization",
      },
      {
        "icon": <ComparisonOutlined
          height={18}
          width={18}
        />,
        "key": "sqle/project/projectID/data-source-comparison",
        "label": <ForwardRef(TypedLink)
          params={
            {
              "projectID": "600300",
            }
          }
          to={
            {
              "path": ":projectID/data-source-comparison",
              "prefix": "/sqle/project",
            }
          }
        >
          结构对比
        </ForwardRef(TypedLink)>,
        "structKey": "data-source-comparison",
      },
    ],
    "label": "SQL开发",
    "permission": undefined,
    "type": "group",
  },
  {
    "children": [
      {
        "icon": <BriefcaseFilled
          height={18}
          width={18}
        />,
        "key": "sqle/project/projectID/exec-workflow",
        "label": <ForwardRef(TypedLink)
          params={
            {
              "projectID": "600300",
            }
          }
          to={
            {
              "path": ":projectID/exec-workflow",
              "prefix": "/sqle/project",
            }
          }
        >
          SQL工单
        </ForwardRef(TypedLink)>,
        "structKey": "exec-workflow",
      },
      {
        "icon": <FileVersionOutlined
          height={18}
          width={18}
        />,
        "key": "sqle/project/projectID/version-management",
        "label": <ForwardRef(TypedLink)
          params={
            {
              "projectID": "600300",
            }
          }
          to={
            {
              "path": ":projectID/version-management",
              "prefix": "/sqle/project",
            }
          }
        >
          版本管理
        </ForwardRef(TypedLink)>,
        "structKey": "version-management",
      },
    ],
    "label": "SQL上线",
    "permission": undefined,
    "type": "group",
  },
  {
    "children": [
      {
        "icon": <PipelineOutlined
          height={18}
          width={18}
        />,
        "key": "sqle/project/projectID/pipeline-configuration",
        "label": <ForwardRef(TypedLink)
          params={
            {
              "projectID": "600300",
            }
          }
          to={
            {
              "path": ":projectID/pipeline-configuration",
              "prefix": "/sqle/project",
            }
          }
        >
          流水线配置
        </ForwardRef(TypedLink)>,
        "structKey": "pipeline-configuration",
      },
    ],
    "label": "CI/CD集成",
    "permission": undefined,
    "type": "group",
  },
  {
    "children": [
      {
        "icon": <ManagementFilled
          height={18}
          width={18}
        />,
        "key": "sqle/project/projectID/sql-management",
        "label": <ForwardRef(TypedLink)
          params={
            {
              "projectID": "600300",
            }
          }
          to={
            {
              "path": ":projectID/sql-management",
              "prefix": "/sqle/project",
              "query": "instance_id&source",
            }
          }
        >
          SQL管控
        </ForwardRef(TypedLink)>,
        "structKey": "sql-management",
      },
      {
        "icon": <PlanFilled
          height={18}
          width={18}
        />,
        "key": "sqle/project/projectID/sql-management-conf",
        "label": <ForwardRef(TypedLink)
          params={
            {
              "projectID": "600300",
            }
          }
          to={
            {
              "path": ":projectID/sql-management-conf",
              "prefix": "/sqle/project",
            }
          }
        >
          SQL管控配置
        </ForwardRef(TypedLink)>,
        "structKey": "sql-management-conf",
      },
    ],
    "label": "SQL管控",
    "permission": undefined,
    "type": "group",
  },
  {
    "type": "divider",
  },
  {
    "children": [
      {
        "icon": <DatabaseFilled
          height={18}
          width={18}
        />,
        "key": "project/projectID/db-services",
        "label": <ForwardRef(TypedLink)
          params={
            {
              "projectID": "600300",
            }
          }
          to={
            {
              "path": ":projectID/db-services",
              "prefix": "/project",
              "query": "address",
            }
          }
        >
          数据源
        </ForwardRef(TypedLink)>,
        "structKey": "instance",
      },
      {
        "icon": <ProfileSquareFilled
          fill="none"
          height={18}
          width={18}
        />,
        "key": "sqle/project/projectID/rule/template",
        "label": <ForwardRef(TypedLink)
          params={
            {
              "projectID": "600300",
            }
          }
          to={
            {
              "path": ":projectID/rule/template",
              "prefix": "/sqle/project",
            }
          }
        >
          审核规则模板
        </ForwardRef(TypedLink)>,
        "structKey": "rule-template",
      },
      {
        "icon": <WorkflowFilled
          height={18}
          width={18}
        />,
        "key": "sqle/project/projectID/progress",
        "label": <ForwardRef(TypedLink)
          params={
            {
              "projectID": "600300",
            }
          }
          to={
            {
              "path": ":projectID/progress",
              "prefix": "/sqle/project",
            }
          }
        >
          审批流程模板
        </ForwardRef(TypedLink)>,
        "structKey": "workflow-template",
      },
      {
        "icon": <MemberFilled
          height={18}
          width={18}
        />,
        "key": "project/projectID/member",
        "label": <ForwardRef(TypedLink)
          params={
            {
              "projectID": "600300",
            }
          }
          to={
            {
              "path": ":projectID/member",
              "prefix": "/project",
            }
          }
        >
          成员与权限
        </ForwardRef(TypedLink)>,
        "structKey": "member",
      },
      {
        "icon": <GearFilled
          height={18}
          width={18}
        />,
        "key": "sqle/project/projectID/push-rule",
        "label": <ForwardRef(TypedLink)
          params={
            {
              "projectID": "600300",
            }
          }
          to={
            {
              "path": ":projectID/push-rule",
              "prefix": "/sqle/project",
            }
          }
        >
          推送规则
        </ForwardRef(TypedLink)>,
        "structKey": "push-rule-configuration",
      },
      {
        "icon": <ResolveFileFilled
          height={18}
          width={18}
        />,
        "key": "sqle/project/projectID/whitelist",
        "label": <ForwardRef(TypedLink)
          params={
            {
              "projectID": "600300",
            }
          }
          to={
            {
              "path": ":projectID/whitelist",
              "prefix": "/sqle/project",
            }
          }
        >
          审核SQL例外
        </ForwardRef(TypedLink)>,
        "structKey": "whitelist",
      },
      {
        "icon": <ExceptionFileOutlined
          height={18}
          width={18}
        />,
        "key": "sqle/project/projectID/sql-management-exception",
        "label": <ForwardRef(TypedLink)
          params={
            {
              "projectID": "600300",
            }
          }
          to={
            {
              "path": ":projectID/sql-management-exception",
              "prefix": "/sqle/project",
            }
          }
        >
          管控SQL例外
        </ForwardRef(TypedLink)>,
        "structKey": "sql-management-exception",
      },
    ],
    "label": "项目配置",
    "permission": undefined,
    "type": "group",
  },
  {
    "type": "divider",
  },
  {
    "children": [
      {
        "icon": <OperateAuditFilled
          height={18}
          width={18}
        />,
        "key": "sqle/project/projectID/operation-record",
        "label": <ForwardRef(TypedLink)
          params={
            {
              "projectID": "600300",
            }
          }
          to={
            {
              "path": ":projectID/operation-record",
              "prefix": "/sqle/project",
            }
          }
        >
          SQLE操作记录
        </ForwardRef(TypedLink)>,
        "permission": "page:operation_record",
        "structKey": "sqle-log",
      },
    ],
    "label": "操作与审计",
    "permission": "page:operation_record",
    "type": "group",
  },
]
`;
